# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../mac/gcc44.patch
# Copyright (C) 2010 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -Nur mac-3.99-u4-b5-orig/src/MACLib/APELink.cpp mac-3.99-u4-b5/src/MACLib/APELink.cpp
--- mac-3.99-u4-b5-orig/src/MACLib/APELink.cpp	2010-08-11 16:17:26.472081342 +0000
+++ mac-3.99-u4-b5/src/MACLib/APELink.cpp	2010-08-11 16:23:27.071079896 +0000
@@ -63,10 +63,10 @@
     if (pData != NULL)
     {
         // parse out the information
-        char * pHeader = strstr(pData, APE_LINK_HEADER);
-        char * pImageFile = strstr(pData, APE_LINK_IMAGE_FILE_TAG);
-        char * pStartBlock = strstr(pData, APE_LINK_START_BLOCK_TAG);
-        char * pFinishBlock = strstr(pData, APE_LINK_FINISH_BLOCK_TAG);
+        const char * pHeader = strstr(pData, APE_LINK_HEADER);
+        const char * pImageFile = strstr(pData, APE_LINK_IMAGE_FILE_TAG);
+        const char * pStartBlock = strstr(pData, APE_LINK_START_BLOCK_TAG);
+        const char * pFinishBlock = strstr(pData, APE_LINK_FINISH_BLOCK_TAG);
 
         if (pHeader && pImageFile && pStartBlock && pFinishBlock)
         {
@@ -81,7 +81,7 @@
                 
                 // get the path
                 char cImageFile[MAX_PATH + 1]; int nIndex = 0;
-                char * pImageCharacter = &pImageFile[strlen(APE_LINK_IMAGE_FILE_TAG)];
+                const char * pImageCharacter = &pImageFile[strlen(APE_LINK_IMAGE_FILE_TAG)];
                 while ((*pImageCharacter != 0) && (*pImageCharacter != '\r') && (*pImageCharacter != '\n'))
                     cImageFile[nIndex++] = *pImageCharacter++;
                 cImageFile[nIndex] = 0;
